packages/mds-utils/tests/state-transition-expected.ts

```diff
@@ -51,7 +67,15 @@ export const expectedTransitions: {
     trip_enter: false,
     trip_leave: false,
     trip_end: false,
-    deregister: true
+    deregister: true,
+    service_enter: false,
+    service_leave: false,
+    depot_enter: true,
+    depot_leave: false,
+    reserve_enter: false,
+    reserve_stop: false,
+    trip_stop: false,
+    trip_resume: false
```

Comment:
I wonder if this would be easier to parse/maintain if it were an array of events, so something like:
```ts
service_end: ['deregister', 'depot_enter']
```
And then, rather than:
```ts
if (expectedTransitions['service_end]['deregister']) {...}
```
You could use:
```ts
if (expcetedTranstions['service_end'].includes('deregister')) {...}
```
Or something like that.

Reply:
Agreed. FWIW as we start to rework the state machine (especially in the context of the reconciliation efforts), this model will probably not be super useful and will need to be significantly reworked.